python/version_details/version_details.py

```diff
@@ -370,6 +371,23 @@ def _set_notes_are_selectable(self, state):
         _set_notes_are_selectable,
     )
 
+    def _get_pre_submit_callback(self):
+        """
+        The pre-submit callback function, if one is registered. If so, this
+        Python callable will be run prior to Note or Reply submission, and
+        will be given the calling :class:`NoteInputWidget` as its first and
+        only argument.
```

Comment:
This diff came in from some other PR, I'm comparing against Jeff's markup attachments PR which doesn't have it.